packages/compile-solidity/src/normalizeOptions.ts

```diff
@@ -10,10 +10,16 @@ export const normalizeOptions = (options: Config) => {
   options.compilers.solc.settings.evmVersion =
     options.compilers.solc.settings.evmVersion ||
     options.compilers.solc.evmVersion;
-  options.compilers.solc.settings.optimizer =
-    options.compilers.solc.settings.optimizer ||
-    options.compilers.solc.optimizer ||
-    {};
+
+  if (options.compilers.solc.disableDefaults) {
+    options.compilers.solc.settings.optimizer =
+      options.compilers.solc.settings.optimizer || {};
+  } else {
+    options.compilers.solc.settings.optimizer =
+      options.compilers.solc.settings.optimizer ||
+      options.compilers.solc.optimizer ||
+      {};
+  }
```

Comment:
This requires a comment and possibly a refactor to help the reader track those object chains in this stanza. Maybe something of this form.

```suggestion

  // Helpful comment...
  const solcOptimizerDefaults = options.compilers.solc.disableDefaults
    ? {}
    : options.compilers.solc.optimizer || {};
  options.compilers.solc.settings.optimizer = options.compilers.solc.settings.optimizer ?? solcOptimizerDefaults;
```

Reply:
I actually think the above is significantly less readable. I am adding a comment though, thank you for the suggestion!